ReactSkia/components/RSkComponentText.cpp

```diff
@@ -51,84 +52,135 @@ inline int getNumberOfLines(std::vector<LineMetrics>& metrics, int maxNumberOfLi
+        parent->expectedAttachmentCount += data.layoutManager->buildParagraph(props.backgroundColor, data.attributedString,
                                                             props.paragraphAttributes,
                                                             true,
                                                             parent->paraBuilder);
         auto paragraph = parent->paraBuilder->Build();
         parent->currentAttachmentCount++;
         if(!parent->expectedAttachmentCount || (parent->expectedAttachmentCount == parent->currentAttachmentCount)) {
             auto frame = parent->getAbsoluteFrame();
-            paragraph->layout(frame.size.width);
+            paragraph->layout(frame.size.width - (2*rectStrokeWidth));
             paragraph->getLineMetrics(metrics);
 
-            numberOfLines = getNumberOfLines(metrics, props.paragraphAttributes.maximumNumberOfLines, frame.size.height);
-            if (numberOfLines){
-                if (EllipsizeMode::Tail == props.paragraphAttributes.ellipsizeMode) {
+            if (rectStrokeWidth) {
+                numberOfLines = getLines(metrics, frame.size.height - (2*rectStrokeWidth));
+            }
+            if (props.paragraphAttributes.maximumNumberOfLines) {
+                numberOfLines = getMaxNumberOfLines(metrics, props.paragraphAttributes.maximumNumberOfLines, frame.size.height);
+                if ((EllipsizeMode::Tail) == (props.paragraphAttributes.ellipsizeMode)) {
                     paraStyle.setEllipsis(u"\u2026");
                 }
-                paraStyle.setMaxLines(numberOfLines);
-                parent->paraBuilder->setParagraphStyle(paraStyle);
+            }
+
+            paraStyle.setMaxLines(numberOfLines);
+            parent->paraBuilder->setParagraphStyle(paraStyle);
+            if (numberOfLines) {
                 paragraph = parent->paraBuilder->Build();
-                paragraph->layout(frame.size.width);
+                paragraph->layout(frame.size.width - (2*rectStrokeWidth));
             }
-            yOffset = yPosOffset(data.attributedString, paragraph->getHeight(), frame);
-            paragraph->paint(canvas, frame.origin.x, frame.origin.y + yOffset);
+
+            if ((props.backgroundColor) || (props.textAttributes.lineHeight >= 0)) {
+                canvas->drawRect(SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height), convertTextColor(props.backgroundColor));
+                canvas->clipRect(SkRect::MakeXYWH(frame.origin.x,frame.origin.y,frame.size.width,frame.size.height));
+            }
+            yOffset = yPosOffset(data.attributedString, paragraph->getHeight(), frame.size.height - rectStrokeWidth);
+            paragraph->paint(canvas, frame.origin.x + rectStrokeWidth, frame.origin.y + yOffset + rectStrokeWidth);
+            RSkDrawUtils::drawBorder(canvas, frame, borderMetrics, props.backgroundColor, 1);
```

Comment:
use using namespace RSkDrawUtils; to avoid specifying namespace explicitly. Refer [View & Image component]

Reply:
Yes